components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -27,7 +27,10 @@
 
 namespace ntp_background_images {
 
-// static
+// BEGIN - static
+base::ObserverList<ViewCounterService::Observer>::Unchecked
```

Comment:
I'm not understanding why this is static, would be great to understand. Each profile has a different ViewCounterService and would accumulate independent counts of NTP views and branded NTP views. Unless we usually just send accumulated totals of other stats for all profiles to p3a?

Reply:
@petemill yup- we're using local_state for P3A, which is profile independent. So this will do the counts regardless of the profile used. Important to note that it shouldn't be incrementing for private tabs

Reply:
I would advise against having a global - first of all, this is discouraged by CC (it should either leaking or trivial - i.e. without a destructor). And second, I think that we generally can ignore different profiles since 87% of users have only one window at a time, so profiles mostly do not interfere.
This way, we can drop the global tracker and just inject some utility code right here

Reply:
Reworked and moved P3A code to `ViewCounterService`, which I think makes more sense. Gets rid of observer complexity and makes service have the ownership 😄 Also - this avoids adding a global 😛